### ISSUE TITLE
System.DllNotFoundException: Microsoft Visual C++ 2017 Redistributable (x64)

### ISSUE THREAD

**Issue by C-hrl** [Trombov/FutureNNAimbot#31]
I get this exception when I launch the .exe and when the "trained" folder has files in it,
I have CUDA 10.0 and VC Redist 2015-2019 which should work as 2017 but isn't detected,

I can't have both installed (if I try to install one of the version while to other is already installed, I get an error message telling me that it's already installed).
If I completely remove 2015-2019 and replace it with 2017 most of my games don't work anymore.

Was wondering how to remove the check for VC 2017 or how to make it understand that 2015-2019 should work like 2017.
Would also be great if FNN used a newer version of Alturos.Yolo as I saw over there that they check for 2015-2019 now.

**Comment by C-hrl** [comment]
After further tests, it seems that the YoloWrapper in Alturos.Yolo is the reason why it doesn't work.
It has a function to look for vc redist 2017.

I found a brutish way of fixing this by decompiling the Alturos.Yolo and setting 
environmentReport.MicrosoftVisualCPlusPlus2017RedistributableExists = true;
line 138 as well as deleting any check for vc redist 2017 in the code
and then recompiling it with these fixes

Hope this help people with similar issues

New dll to get rid of vc redist error (don't forget to back up your original file, may not work with future updates of FNN)
[VCRedistCorrectedDLL.zip](https://github.com/Trombov/FutureNNAimbot/files/4970787/VCRedistCorrectedDLL.zip)

**Comment by Neakita** [comment]
Hi, could you send me your corrected version?

**Comment by yosoloman** [comment]
same couuld you send

**Comment by yosoloman** [comment]
ca you send me the dll

**Comment by KittenTheMeow** [comment]
> ca you send me the dll

**Comment by Neakita** [comment]
Can't you download it from here?

**Comment by Neakita** [comment]
https://drive.google.com/file/d/1KZxIwyfZBkTxDctcDKdJKS-6AxnKKnki/view?usp=sharing

**Comment by Neakita** [comment]
I edited the code through visual studio, updated NuGet, installed a Redist of an older version, I was written that the video card was not detected, I deleted the check for this, it started, but the speed was not very fast, as if it worked on the processor.

**Comment by yosoloman** [comment]
trhanks

**Comment by Neakita** [comment]
> I edited the code through visual studio, updated NuGet, installed a Redist of an older version, I was written that the video card was not detected, I deleted the check for this, it started, but the speed was not very fast, as if it worked on the processor.

Yeah, it working on CPU.
![image](https://user-images.githubusercontent.com/68691636/88322599-64227b00-cd3a-11ea-9dcb-51873ca3cd14.png)

**Comment by yosoloman** [comment]
i had to request access

**Comment by yosoloman** [comment]
plz let me have access so i can download

**Comment by Neakita** [comment]
Am.. u havent access?

**Comment by Neakita** [comment]
Done.

**Comment by yosoloman** [comment]
thxz

**Comment by yosoloman** [comment]
i still get the issue of it not recognizing the vc runtime 2017

**Comment by Neakita** [comment]
With not modified dll - yes, on moded dll it working on cpu.

**Comment by yosoloman** [comment]
i have the correct cuda kit and have vc 2015-2019

**Comment by yosoloman** [comment]
Unhandled Exception: System.DllNotFoundException: Microsoft Visual C++ 2017 Redistributable (x64)
   at Alturos.Yolo.YoloWrapper.Initialize(String configurationFilename, String weightsFilename, String namesFilename, Int32 gpu, Boolean ignoreGpu)
   at FutureNNAimbot.Program.Main(String[] args)

**Comment by yosoloman** [comment]
i dont understand why this wont work even with the patched dll

**Comment by Neakita** [comment]
How did you decompile the DLL?

**Comment by yosoloman** [comment]
is there a fix to not look for the vc 2017 which is replaced by the 2015-2019 update

**Comment by yosoloman** [comment]
i didnt decompile

**Comment by yosoloman** [comment]
i thot yor dll was the fizxed one

**Comment by Neakita** [comment]
"I found a brutish way of fixing this by decompiling the Alturos.Yolo and setting"

**Comment by yosoloman** [comment]
can you give me the fix

**Comment by yosoloman** [comment]
i dnot know how to fix

**Comment by yosoloman** [comment]
its just the one i donwloaded from you

**Comment by Neakita** [comment]
•-•

I can, but it working on cpu, i trying to fix it.

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88323623-f4ad8b00-cd3b-11ea-8032-de5d16faa000.png)

**Comment by yosoloman** [comment]
i havent leanred much coding to be able to implement this. if I get a newer cuda version will it work for the dll you sent

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88323723-19096780-cd3c-11ea-978b-99fe84c57342.png)
But i not programmer.

**Comment by yosoloman** [comment]
where is it that you are writing that code to

**Comment by Neakita** [comment]
I installed 10.1 cuda coz new version of nugget need it.

**Comment by yosoloman** [comment]
oh ok

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88323838-45bd7f00-cd3c-11ea-87ea-d8cfb05bb3d5.png)

**Comment by yosoloman** [comment]
so if i get that should it fix since I have your dll

**Comment by yosoloman** [comment]
i ahve a 2080ti so i should be compatible

**Comment by KittenTheMeow** [comment]
I have 1660s :D

**Comment by Neakita** [comment]
So, i'll send it?

**Comment by yosoloman** [comment]
ok

**Comment by yosoloman** [comment]
what steps do i need to take to get this to work

**Comment by Neakita** [comment]
https://drive.google.com/file/d/1SDCdw083gJ1x36tSzhNGMOfEILJC394R/view?usp=sharing

**Comment by Neakita** [comment]
I tried everything that comes into my head, in particular, I have an older version of the C++Redist

**Comment by yosoloman** [comment]
Unhandled Exception: System.MissingMethodException: Method not found: 'Void Alturos.Yolo.YoloWrapper..ctor(System.String, System.String, System.String, Int32, Boolean)'.
   at FutureNNAimbot.Program.Main(String[] args)

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88324270-e57b0d00-cd3c-11ea-9b7f-0d1bb6bf1c6e.png)

**Comment by yosoloman** [comment]
now i only have this error but the other one is fixed

**Comment by Neakita** [comment]
> Unhandled Exception: System.MissingMethodException: Method not found: 'Void Alturos.Yolo.YoloWrapper..ctor(System.String, System.String, System.String, Int32, Boolean)'.
> at FutureNNAimbot.Program.Main(String[] args)

Am... I need to think about it.

**Comment by yosoloman** [comment]
i have the 14.26 version do i need to downgrade to .25

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88324395-178c6f00-cd3d-11ea-9d22-76666b88b9a0.png)
I'm confused about the variable "Ignore gpu"

**Comment by yosoloman** [comment]
is there a way you can just send me all of the files you have for the cheat

**Comment by yosoloman** [comment]
i trained the game for apex now its just when I run it after it wont work

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88324518-3b4fb500-cd3d-11ea-9d04-359b879ab370.png)
I can send it, ha-hah.

**Comment by yosoloman** [comment]
lol

**Comment by yosoloman** [comment]
im runnign  the exe version how do you run yours

**Comment by Neakita** [comment]
U coder?

**Comment by yosoloman** [comment]
no im not im starting some python will hopefully eventually

**Comment by Neakita** [comment]
There is an executable file there

**Comment by yosoloman** [comment]
oh ok

**Comment by Neakita** [comment]
I need to start it on gpu, else it will work very slow and my cpu will burn

**Comment by yosoloman** [comment]
alright, i gotta go to bed soon so if you can just upload your files in a zip or google drive I will download them and paste my trained files in or retrain if that doesn't work. thanks much appreciated

**Comment by Neakita** [comment]
K :)

**Comment by yosoloman** [comment]
somebody siad banned in apex but i dont understand how that's possible because basically it is a motion slash image type tracker it doesn't hack the game memory to get play coordinates so it really is hard to ban

**Comment by yosoloman** [comment]
thanks ill download and it should work I'm run ing i78700k with 2080ti so should be compatible no problem

**Comment by Neakita** [comment]
> somebody siad banned in apex but i dont understand how that's possible because basically it is a motion slash image type tracker it doesn't hack the game memory to get play coordinates so it really is hard to ban

Some anticheat can monitoring winapi, mouse_event.

**Comment by Neakita** [comment]
FACE IT

> > somebody siad banned in apex but i dont understand how that's possible because basically it is a motion slash image type tracker it doesn't hack the game memory to get play coordinates so it really is hard to ban
> 
> Some anticheat can monitoring winapi, mouse_event.

**Comment by yosoloman** [comment]
true

**Comment by yosoloman** [comment]
it can detec tif you hit every shot or impossible input but other than that if your careful you can be clean

**Comment by Neakita** [comment]
Does apex have a reporting system?
If true, it can be reason.

**Comment by yosoloman** [comment]
yeah it does

**Comment by yosoloman** [comment]
mainly people who blatantly do it

**Comment by yosoloman** [comment]
how big is taht upload for google drive

**Comment by Neakita** [comment]
My repos weighs < 1gb •o•

**Comment by yosoloman** [comment]
oh so it might be a bit before I can access it lol

**Comment by Neakita** [comment]
Необработанное исключение: Alturos.Yolo.YoloInitializeException: Cudnn not found
   в Alturos.Yolo.YoloWrapper.Initialize(String configurationFilename, String weightsFilename, String namesFilename, GpuConfig gpuConfig)
   в FutureNNAimbot.Program.Main(String[] args) в C:\Users\User\source\repos\FutureNNAimbot\FutureNNAimbot\Program.cs:строка 193

**Comment by yosoloman** [comment]
if it works for you then i should have no problems ill try with current cuda and upgrade if needed. I'm just hoping I can paste my dark files and trained files

**Comment by yosoloman** [comment]
is that russian?

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88325952-fb89cd00-cd3e-11ea-9e13-ebb961cf5588.png)
Yep, russian.

**Comment by yosoloman** [comment]
so does it work on gpu for you then or are you getting same problem as me

**Comment by Neakita** [comment]
Without "gpuConfig" it working, with - no.

**Comment by Neakita** [comment]
> so does it work on gpu for you then or are you getting same problem as me

True.

**Comment by yosoloman** [comment]
ok yeah ill just get the your files then because I dont know how to fix mine

**Comment by Neakita** [comment]
Compiled?

**Comment by yosoloman** [comment]
well just whatever your folder is like in a google drive upload with all the files like that repository screenshot unless it doesn't work

**Comment by yosoloman** [comment]
the dowload files dont work frome the guy who made it

**Comment by yosoloman** [comment]
maybe the c sharp ones do but I dont know how to make it it into an exe when I tried it was being weird

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88326903-f8431100-cd3f-11ea-9d1b-d81cd87b5060.png)

**Comment by Neakita** [comment]
I will send a repository, if you can run it on the GPU, let me know.

**Comment by yosoloman** [comment]
Ok

**Comment by yosoloman** [comment]
Sorry I was confused were you able to run on gpu 

Just not CPU correct?

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88327215-45bf7e00-cd40-11ea-9ebc-a3a27617a586.png)
Wow....

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88327290-5c65d500-cd40-11ea-9891-92f01edcba35.png)
Without debug build.

**Comment by yosoloman** [comment]
Ok

**Comment by Neakita** [comment]
https://drive.google.com/file/d/1oFsQ9mN96jcEIuS1vWvY0D33D2oChP9E/view?usp=sharing

**Comment by Neakita** [comment]
I need to try it.
![image](https://user-images.githubusercontent.com/68691636/88328235-8ff52f00-cd41-11ea-8f55-52cfa1340c4b.png)

**Comment by yosoloman** [comment]
K I'll try it later today thanks

**Comment by Neakita** [comment]
It didn't help me.

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88330079-4d812180-cd44-11ea-950a-615265de9ef3.png)
Something interesting.

**Comment by Neakita** [comment]
![image](https://user-images.githubusercontent.com/68691636/88330193-7c979300-cd44-11ea-9c0a-94db4184cf3e.png)

**Comment by Neakita** [comment]
I just now saw the requirements for the GPU, it was written that you need to install, in particular, version 10.2 specifically, I followed the instructions by deleting the old versions, and everything worked, now the frame is drawn quickly enough, it should be and it will work well... lol.
![image](https://user-images.githubusercontent.com/68691636/88335469-cc7a5800-cd4c-11ea-8670-c2b2cde8f1f0.png)

**Comment by yosoloman** [comment]
K I will work on this tomorrow and let you know. Thanks 👍

**Comment by C-hrl** [comment]
Waw, I just log on my email and 115 mails 0_0 (I hate you github!)
[VCRedistCorrectedDLL.zip](https://github.com/Trombov/FutureNNAimbot/files/4970781/VCRedistCorrectedDLL.zip)
This is the corrected version of the dll, unzip and replace the original one with this one
Also you can use a program like dotPeek to edit dll's but if you are not familiar with programming I wouldn't recommend you to tweak these kind of files too much :)
Also please, next time refrain yourself from talking that much, just look at other opened issues, they never go beyond 10 messages, thanks.

**Comment by yosoloman** [comment]
which do i run kittenthemeow

**Comment by yosoloman** [comment]
theres  more that one exe and if I run the first one it has vc studio error and the others have the error of not finding the game overlay

**Comment by yosoloman** [comment]
also do i replace the cunn file in the x64 file

**Comment by yosoloman** [comment]
also @cbenimed i get this error when i replace the dll 



Unhandled Exception: System.TypeLoadException: Could not load type 'Alturos.Yolo.GpuConfig' from assembly 'Alturos.Yolo, Version=2.4.1.0, Culture=neutral, PublicKeyToken=null'.
   at FutureNNAimbot.Program.Main(String[] args)

**Comment by yosoloman** [comment]
i cant get it to work can you upload your final fixed version because I still get errors and I don't understand why

**Comment by C-hrl** [comment]
@yosoloman replace to cudnn file in the x64 file ? I may be wrong but there was nothing that needed replacement there.
I just edited 
![image](https://user-images.githubusercontent.com/55942930/88453274-c4620b80-ce65-11ea-9ad1-6317ab2cb52b.png)
Also remember that you can edit your posts instead of writing 5 consecutive comments ;)
Also do you have a Nvidia GPU ?
As far as I remember you just had to download cudnn 10.0, install it and that's it.
![image](https://user-images.githubusercontent.com/55942930/88453338-4fdb9c80-ce66-11ea-8034-a0f2ee75bce6.png)
There is nothing to replace in x64
Here is a back up of the yolo files in x64 if you replaced them, The cudnn file in x64 is too large to send on github so you'll have to download it from the repo
[x64yolodllbackup.zip](https://github.com/Trombov/FutureNNAimbot/files/4975183/x64yolodllbackup.zip)

**Comment by yosoloman** [comment]
the fixz is to copy al the main files into trhe same folder as the exe file

**Comment by yosoloman** [comment]
all the gpu dll and everything

**Comment by C-hrl** [comment]
No, the fix is to replace arturos.yolo.dll with the one provided in the second post of this thread. I think you said that you replaced cudnn in x64 but you musn't replace that (if you did I provided the two gpu dll that must go in the x64 folder).
I don't know what you are doing and just edit your comments instead of writing 2 separate comments for nothing. This thread is supposed to help people code, not talk. I provided my solution, if it doesn't work I can't help you.
There are no files that need to be renamed or anything, just switch the original arturos.yolo.dll with the one in the first zip file(you must unzip it first of course).
I recommend you to download a clean version of FNN without the edits of KittenTheMeow and just replace Arturos.yolo.dll with the one I made. This thread is just here to fix the vs redist bug.
I can't help you with bugs that arise because you modified the gpuconfig which is has nothing to do with the vs redist error! If you get another error than the redist one after doing what I said check if there is already an opened or closed thread about the bug and if not open a new one. Now stop necroing old threads please :)

![image](https://user-images.githubusercontent.com/55942930/88455513-21fe5400-ce76-11ea-98e2-42cff31bb888.png)
yes my file actually just does that

KittenTheMeow actually modified the thing to work with an older redist (or newer?) which is just making everything harder to do, FNN works better with vs redist 2015-2019 but has a dumb check looking specifically for 2017. 2017 is included in 2015-2019 so you MUST NOT do what KittenTheMeow did or you will have more work to do (and it won't use the GPU for computation). My version just removed the check for 2017 so as long as you have vs redist 2015-2019 it will work as intended without needing to replace anything else, don't change cudnn, don't install an older version of vs redist (will break most of your games), just download or restore your back up of the original FNN and replace the file 
![image](https://user-images.githubusercontent.com/55942930/88455466-cb911580-ce75-11ea-8d4c-c32c24f07a8f.png)
with the one in this zip
![image](https://user-images.githubusercontent.com/55942930/88455541-56721000-ce76-11ea-85b1-e607659e89e5.png)

So to recap, install the cudnn version the author said to install, download FNN and replace the file with mine. Now it should work on your GPU and you can start training it.
